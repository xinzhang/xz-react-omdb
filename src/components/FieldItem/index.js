import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const StyledItem = styled.div`
  fontsize: '14px';
  padding: 4px 0px;
`;

const FieldItem = ({ field, text }) => {
  return text ? (
    <StyledItem>
      <b>{field}</b>: {text}
    </StyledItem>
  ) : null;
};

FieldItem.prototype = {
  field: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
};

export default FieldItem;
