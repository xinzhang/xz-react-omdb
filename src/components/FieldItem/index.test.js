import React from 'react';
import renderer from 'react-test-renderer';
import FieldItem from '.';

describe('<FieldItem /> Component', () => {
  test('it should match the FieldItem with value', () => {
    const component = renderer.create(<FieldItem field="name" text="value" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('it should match the FieldItem without value', () => {
    const component = renderer.create(<FieldItem field="name" text="" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
