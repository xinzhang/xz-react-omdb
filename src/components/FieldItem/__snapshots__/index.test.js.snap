// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FieldItem /> Component it should match the FieldItem with value 1`] = `
<div
  className="sc-AxjAm bTkFZh"
>
  <b>
    name
  </b>
  : 
  value
</div>
`;

exports[`<FieldItem /> Component it should match the FieldItem without value 1`] = `null`;
